src/ranged.cpp

```diff
@@ -266,26 +251,41 @@ bool player::handle_gun_damage( item &it )
         }
     }
     if( curammo_effects.count( "BLACKPOWDER" ) ) {
-        if( !it.faults.count( fault_gun_blackpowder ) &&
-            it.faults_potential().count( fault_gun_blackpowder ) ) {
-            it.faults.insert( fault_gun_blackpowder );
-        }
         if( ( it.ammo_data()->ammo->recoil < firing->min_cycle_recoil ) &&
-            it.faults_potential().count( fault_gun_chamber_spent ) ) {
+            it.faults_potential().count( fault_gun_chamber_spent ) && ( it.type->gun->ups_charges < 1 ) ) {
```

Comment:
No need to check for UPS charges here; whether a gun will cycle or not is already figured out via the "min_cycle_recoil" stat on it; guns that use external power to cycle would simply have it set to 0 and so would cycle with any ammo.

Reply:
OK, I'll take your word for it. Seems to be working correctly without it for a minigun at least.